eslint.config.js

```diff
@@ -72,10 +72,10 @@ const allFilesConfig = tsEslint.config({
  */
 export function getTypeScriptProjectRules(root) {
   return tsEslint.config({
-    files: ["**/packages/*/src/**/*.ts", "**/packages/*/src/**/*.tsx"],
+    files: ["**/packages/*/src/**/*.ts", "**/packages/*/src/**/*.tsx", "**/packages/http-client-java/emitter/src/**/*.ts"],
```

Comment:
lint job throws below error when running `pnpm run lint`:
https://github.com/microsoft/typespec/actions/runs/11267081108/job/31331611206

Run pnpm run lint

> @typespec/monorepo@0.0.1 lint /home/runner/work/typespec/typespec
> eslint . --max-warnings=0


Oops! Something went wrong! :(

ESLint: 9.11.1

RangeError: Maximum call stack size exceeded
Occurred while linting /home/runner/work/typespec/typespec/packages/http-client-java/emitter/src/common/client.ts:31
Rule: "@typescript-eslint/no-deprecated"
    at findBaseOfDeclaration (/home/runner/work/typespec/typespec/node_modules/.pnpm/typescript@5.[6](https://github.com/microsoft/typespec/actions/runs/11267081108/job/31331611206#step:6:7).2/node_modules/typescript/lib/typescript.js:149319:31)
    at getJsDocTagsOfDeclarations (/home/runner/work/typespec/typespec/node_modules/.pnpm/typescript@5.6.2/node_modules/typescript/lib/typescript.js:149282:29)
    at SymbolObject.getContextualJsDocTags (/home/runner/work/typespec/typespec/node_modules/.pnpm/typescript@5.6.2/node_modules/typescript/lib/typescript.js:149109:44)
    at /home/runner/work/typespec/typespec/node_modules/.pnpm/typescript@5.6.2/node_modules/typescript/lib/typescript.js:14928[7](https://github.com/microsoft/typespec/actions/runs/11267081108/job/31331611206#step:6:8):27
    at /home/runner/work/typespec/typespec/node_modules/.pnpm/typescript@5.6.2/node_modules/typescript/lib/typescript.js:14932[8](https://github.com/microsoft/typespec/actions/runs/11267081108/job/31331611206#step:6:9):21
    at firstDefined (/home/runner/work/typespec/typespec/node_modules/.pnpm/typescript@5.6.2/node_modules/typescript/lib/typescript.js:2307:20)
    at findBaseOfDeclaration (/home/runner/work/typespec/typespec/node_modules/.pnpm/typescript@5.6.2/node_modules/typescript/lib/typescript.js:14[9](https://github.com/microsoft/typespec/actions/runs/11267081108/job/31331611206#step:6:10)324:[10](https://github.com/microsoft/typespec/actions/runs/11267081108/job/31331611206#step:6:11))
    at getJsDocTagsOfDeclarations (/home/runner/work/typespec/typespec/node_modules/.pnpm/typescript@5.6.2/node_modules/typescript/lib/typescript.js:[14](https://github.com/microsoft/typespec/actions/runs/11267081108/job/31331611206#step:6:15)9282:29)
    at SymbolObject.getContextualJsDocTags (/home/runner/work/typespec/typespec/node_modules/.pnpm/typescript@5.6.2/node_modules/typescript/lib/typescript.js:149109:44)
    at /home/runner/work/typespec/typespec/node_modules/.pnpm/typescript@5.6.2/node_modules/typescript/lib/typescript.js:149287:27
 ELIFECYCLE  Command failed with exit code 2.v

Reply:
But if we only include files under `src/` folder, lint will pass. I guess it's because eslint can't handle too much levels of folder hierarchy.
see this commit: 82cfd66ad5af0e79720745e8e7f61dd0cc2b39ed
job: https://github.com/microsoft/typespec/actions/runs/11267081108/job/31331611206

Reply:
@timotheeguerin May I know your suggestion on this? Basically we want to add deprecation lint rules for ts files under `http-client-java/emitter/`, but when we include those ts files, it will throw `Maximum call stack size exceeded` error, seems due to eslint can't handle too much levels of folder hierarchy.

Reply:
yeah the typescript eslint dependency is struggling with large projects

Reply:
Thanks @timotheeguerin for reply! Do you have suggestions on how to add deprecation lint rules for `http-client-java`'s files (I guess other language's emitters also need this)?